README.md
# eatarranza

A collection of my journey as a developer!

## Tech

- ⚡️ Next.js 13
- ⚛️ React 18
- ✨ TypeScript
- 💨 Tailwind CSS 3


CHANGELOG.md
<!-- //!STARTERCONF Remove this file, this is used as the starter changelog -->

# ts-nextjs-tailwind-starter changelog

This changelog is manually generated and not accurate with the package.json, only to show the changes since the last release.

## 1.0.0 (2022-11-12)


### Features

* add themes ([ccf5d01](https://github.com/kielllll/eatarranza/commit/ccf5d01aa1e8a944cd8a7b879a5d43194c6a40fe))
* setup layout and providers ([358a317](https://github.com/kielllll/eatarranza/commit/358a31718479feaa078aa8f565ca03668b03ab0e))

## 0.5.4 - 2022-07-22

### New Features

- #### Release Please

    Standard Version is now deprecated, and ts-nextjs-tailwind-starter is now using release please. Activate them on `.github/workflows/release-please`

### Improvements & Bug Fixes

- #### More Efficient Lint Actions
    
    Lint workflow is now cached and will cancel previous run if there are 2 concurrent runs.

## 0.5.3 - 2022-02-27

### New Features

- #### Shimmer for NextImage and Skeleton
    
    Addition of shimmer & blur placeholder for NextImage, and new Skeleton Component with shimmer effect.

    https://user-images.githubusercontent.com/55318172/155867729-8c3176ad-ede4-4443-b42b-780517615e5a.mp4


- #### Support for SVGR

    You can directly import SVG like

    ```tsx
    import Vercel from '~/svg/Vercel.svg';

    <Vercel className='text-5xl' />
    ```

- #### Public Folder Path Mapping

    Easily access public folder with `~/` prefix.

- #### Tailwind CSS Prettier Sorter

    ts-nextjs-tailwind-starter now use first-party plugin `prettier-plugin-tailwindcss`

### Improvements & Bug Fixes

- #### Layout Declared Twice

    Fix issue where adding elements to Layout ends up rendering them twice

- #### ESLint Curly Brace Rule

    New autofixable rule
    
    ```tsx
    props={'hi'}

    will become

    props='hi'
    ```

## 0.5.2 - 2021-12-30

### New Features

- #### New Component: PrimaryLink

    Add a link component with accent color on top of UnstyledLink.

### Improvements & Bug Fixes

- #### Unused Import ESlint Autofix

    Unused import will automatically be removed by the ESlint autofix.

- #### Renamed CustomLink to UnderlineLink

    This is to compensate the new PrimaryLink component

- #### Primary Button & ButtonLink Shade

    The shade of the button is now using the `500` instead of `400`.


## 0.5.1 - 2021-12-26

### New Features

- #### New Snippets Wrap: clsx and fragment `<></>`
    
    You can select text then wrap it with clsx or React.Fragment shorthand.
    
    https://user-images.githubusercontent.com/55318172/147401848-3db5dba0-ef71-4f25-9f47-c7908beba69e.mp4


## 0.5.0 - 2021-12-21

### New Features

- #### Expansion Pack
    You can easily add expansion such as React Hook Form + Components, Storybook, and more just using a single command line.

    https://user-images.githubusercontent.com/55318172/146631994-e1cac137-1664-4cfe-950b-a96decc1eaa6.mp4

    Check out the [expansion pack repository](https://github.com/theodorusclarence/expansion-pack) for the commands

### Improvements & Bug Fixes

- #### Can't Use Layout Fill on NextImage

    Using layout fill will make the width and height optional

- #### Vertically center Button & ButtonLink

    Adds `items-center` to make the button vertically centered


## 0.4.1 - 2021-12-12

### New Features

- #### Tailwind CSS v3
    
    The color palette configuration is also updated accordingly. 

## 0.4.0 - 2021-12-02

### New Features

- #### Button & ButtonLink Variants

    - New Variant: **Outline** and **Ghost**, you can also add `isDarkBg` prop if you are using these variants with dark background.
    - Animated Underline style on **Primary**, **Dark**, **Light** is removed
    - Added `ring-primary-500` on `focus-visible`

    ![Button Variants](https://user-images.githubusercontent.com/55318172/144385213-632b3e1f-9a0e-4184-82e0-7905ee3318b4.gif)

- #### ArrowLink


    Adds an animated arrow, this component is Polymorphic, the default element is `CustomLink`, you can extend it with `as` prop.

    ```tsx
    <ArrowLink
      as={ButtonLink}
      variant='light'
      href='/'
    >
      Register now
    </ArrowLink>
    ```

    ![Arrow Link Feature](https://user-images.githubusercontent.com/55318172/144385991-f3521d52-e0a8-49c5-8e87-409231fdd5b6.gif)

- #### Change default theme to white

    | ![Home Page](https://user-images.githubusercontent.com/55318172/144386763-00e6c3fd-ee2e-4c9e-87f8-18b036bdc2e1.png) | ![404](https://user-images.githubusercontent.com/55318172/144386764-0e4b4fb0-35a8-4725-a795-f998b06543a1.png) |
    | - | - |

### Improvements & Bug Fixes

- #### Split Next.js Link Props Type

    Now, to add props to Next.js `<Link>` component, you can use `nextLinkProps`.

    ```tsx
    <UnstyledLink 
      href='/'
      nextLinkProps={{
        shallow: true,
      }}
    >
      Link
    </UnstyledLink>
    ```

    The rest of `<a>` props can be directly added as a prop.

- #### Add Motion Safe to Animations

    All components animation respect user preference about motion.

## 0.3.0 - 2021-12-01

### New Features

- #### Create Branch & Auto Resolve Issue Actions

    | ![Create Branch Actions](https://user-images.githubusercontent.com/55318172/144379834-8c3e4d4f-d584-4253-9ad8-b9f1d468ed01.gif) <br> Auto Create Branch | ![Auto Resolve](https://user-images.githubusercontent.com/55318172/144382044-0132e755-9cd5-4805-a756-4086f67b3282.gif) <br> Auto Resolve  |
    | :--: | :--: |

    You have to install the app for your organization/account/repository from the [GitHub Marketplace](https://github.com/marketplace/create-issue-branch) for this to work.

    The branch will be created on **assign** with format `i${number}-${issue_title_lowercase}`.

- #### Custom Tailwind CSS Class Sorter


data/data.js
export const featuredProjects = [
  {
    name: 'Digital Knowmad',
    description:
      'A capstone project during my web development bootcamp with Zuitt.',
    image: '/images/none.jpg',
    repoLink:
      'https://gitlab.com/tuitt/students/batch95/ezekiel-tarranza/csp2/app-client',
    productionLink: 'https://zeke-csp2-app-client.glitch.me/',
    tags: ['html', 'css', 'javascript'],
  },
  // TODO: enable once Budget Wiser has been updated
  // {
  //     name: "Budget Wiser",
  //     description:
  //         "A capstone project during my web development bootcamp with Zuitt.",
  //     image: "/images/none.jpg",
  //     repoLink: "https://gitlab.com/kielllll/csp3-client",
  //     productionLink: "https://csp3-client-iota.vercel.app/",
  //     tags: ["next.js", "html", "css", "vercel"],
  // },
  {
    name: 'Peformance Analysis: TOTP Algorithm',
    description: 'A research project during my final year in college.',
    image: '/images/none.jpg',
    repoLink: 'https://github.com/kielllll/thesis-ui',
    productionLink: 'https://thesis-totp.herokuapp.com/',
    tags: ['nuxt.js', 'html', 'css', 'heroku'],
  },
  {
    name: 'Web Portfolio',
    description:
      "My own web portfolio. Inspired by Benjanmin Carlson's Profile",
    image: '/images/none.jpg',
    repoLink: 'https://github.com/kielllll/web-portfolio',
    productionLink: 'https://zeke-portfolio.vercel.app/',
    tags: ['next.js', 'html', 'css', 'vercel'],
  },
]

export const experiences = [
  {
    company: 'Azoo Inc.',
    role: 'Freelance Developer',
    duration: 'June 2021 - April 2022',
    jobType: ['Full Time'],
    description:
      'I work as a developer in WASIMIL who handles mainly the front-end and sometimes the back-end. It is a hotel management software based in Japan that focuses on helping small to medium sized hotels to properly manage their hotel. My work on WASIMIL revolves around notification-like features, debugging, component development, and other frontend improvements.',
  },
  {
    company: 'Summit Digital',
    role: 'Freelance Developer',
    duration: 'March 2021 - July 2021',
    jobType: ['Part Time'],
    description:
      'I work as a developer in frontend and automations to work on our product in Summit Digital. Summit Digital is an Australian company that focuses on creating lead-generating automations that would help businesses specifically swim schools to gather leads. My role in the team is to generate automations using pipedream or zapier and leadshook for the lead pipeline',
  },
  {
    company: 'itGenius',
    role: 'Software Engineer',
    duration: 'August 2019 - July 2021',
    jobType: ['Full Time'],
    description:
      'I worked in ITGenius as a Developer mainly for custom reporting, scripts, and automations. I am exposed in doing the following task - Chatbot AI (Frontend & Backend), reports using Google Data Studio, custom scripts using Google Apps Script, and workflow automations using Zapier and Integromat.',
  },
  {
    company: 'Concentrix',
    role: 'Technical Support Expert',
    duration: 'June 2019 - August 2019',
    jobType: ['Full Time'],
    description:
      'I worked in Concentrix as Technical Support Representative in a Voice Account. This is where I honed my soft skills, specifically communication.',
  },
  {
    company: 'itGenius',
    role: 'Software Engineer',
    duration: 'October 2018 - February 2019',
    jobType: ['Intern'],
    description:
      'In my internship with ITGenius, I am exposed in doing the following task - fixing bugs in existing codebase, creating reports using Google Data Studio, and custom scripts using Google Apps Script.',
  },
]

export const education = [
  {
    name: 'Zuitt',
    duration: 'February 2021 - June 2021',
    course: 'Web Developer Bootcampee',
    description:
      'In order to hone my skills to further improve my career. I have attended a bootcamp which focuses on MERN stack. We’ve had 3 capstone projects that we have developed.',
  },
  {
    name: 'University of Mindanao',
    duration: 'June 2016 - Present',
    course: 'BS in Computer Science',
    description: 'A lot of Mathematics, Theories, and Principles!',
  },
]

export const hardware = [
  {
    type: 'Processor',
    name: 'AMD Ryzen 5 3600',
  },
  {
    type: 'RAM',
    name: 'Team Elite 2 x 8GB',
  },
  {
    type: 'Graphics Card',
    name: 'Nvidia GTX 1650',
  },
  {
    type: 'SSD',
    name: 'Samsung M.2 500GB, AData SATA 2.5 265GB',
  },
  {
    type: 'HDD',
    name: 'Seagate 1TB',
  },
  {
    type: 'Primary Monitor',
    name: 'Asus VP259QGR 144hz 23.8in',
  },
  {
    type: 'Secondary Monitor',
    name: 'HP 23.8in',
  },
  {
    type: 'Keyboard',
    name: 'TM680 (Modded)',
  },
  {
    type: 'Mouse',
    name: 'ROG Stix Impact II Wireless',
  },
]

export const software = [
  {
    type: 'IDE',
    name: 'Visual Studio Code / WebStorm / PhpStorm',
  },
  {
    type: 'API Tester',
    name: 'Postman / GraphQL Playground',
  },
  {
    type: 'Browser',
    name: 'Google Chrome / Brave',
  },
  {
    type: 'UI/UX Prototyping',
    name: 'Figma',
  },
]


tailwind.config.js
/* eslint-disable @typescript-eslint/no-var-requires */
const { fontFamily } = require('tailwindcss/defaultTheme');

/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ['./src/**/*.{js,jsx,ts,tsx}'],
  darkMode: 'class',
  theme: {
    extend: {
      fontFamily: {
        primary: ['Inter', ...fontFamily.sans],
      },
      colors: {
        primary: {
          // Customize it on globals.css :root
          200: 'rgb(var(--tw-clr-primary-200) / <alpha-value>)',
          300: 'rgb(var(--tw-clr-primary-300) / <alpha-value>)',
          400: 'rgb(var(--tw-clr-primary-400) / <alpha-value>)',
          500: 'rgb(var(--tw-clr-primary-500) / <alpha-value>)',
        },
        dark: '#222222',
        'semi-dark': '#121212',
      },
      keyframes: {
        flicker: {
          '0%, 19.999%, 22%, 62.999%, 64%, 64.999%, 70%, 100%': {
            opacity: 0.99,
            filter:
              'drop-shadow(0 0 1px rgba(252, 211, 77)) drop-shadow(0 0 15px rgba(245, 158, 11)) drop-shadow(0 0 1px rgba(252, 211, 77))',
          },
          '20%, 21.999%, 63%, 63.999%, 65%, 69.999%': {
            opacity: 0.4,
            filter: 'none',
          },
        },
        tilt: {
          '0%, 50%, 100%': {
            transform: 'rotate(0deg)',
          },
          '25%': {
            transform: 'rotate(0.5deg)',
          },
          '75%': {
            transform: 'rotate(-0.5deg)',
          },
        },
      },
      animation: {
        flicker: 'flicker 3s linear infinite',
        tilt: 'tilt 10s infinite linear',
      },
    },
  },
  plugins: [require('@tailwindcss/forms'), require('@tailwindcss/typography')],
};
